### DISCUSSION TITLE
[Bug] Java agent endpoint config not take effect when use @PathVariable

### DISCUSSION THREAD

**Discussion by stalary** [apache/skywalking#11624] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

When I use java agent 9.0, jdk21 spirngboot 3.1.4 spirngmvc 6.0.12, and set SW_PLUGIN_SPRINGMVC_USE_QUALIFIED_NAME_AS_ENDPOINT_NAME=true, but the method who used @PathVariable can not regular work.

### What you expected to happen

@GetMapping("{id}")
public void testId(@PathVariable long id)

endpoint is {id} not long number

### How to reproduce

use java agent 9.0

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
I am not sure what happens, you could debug `AbstractMethodInterceptor#beforeMethod`, which should guide you to go through the logic.
